app/src/main/java/protect/card_locker/ShortcutHelper.java

```diff
@@ -155,4 +155,26 @@ Bitmap createAdaptiveBitmap(@NotNull Bitmap in, int paddingColor) {
                 .setIntent(intent)
                 .setIcon(icon);
     }
+    static ShortcutInfoCompat.Builder createGroupShortcutBuilder(Context context, Group group) {
+        Intent intent = new Intent(context, MainActivity.class);
+        intent.setAction(Intent.ACTION_MAIN);
+        // Prevent instances of the view activity from piling up; if one exists let this
+        // one replace it.
+        intent.setFlags(intent.getFlags() | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        final Bundle bundle = new Bundle();
+        // Done like with the cards but with group._id (a string)
+        bundle.putString("id", group._id);
+        bundle.putBoolean("view", true);
+        intent.putExtras(bundle);
+
+        Bitmap iconBitmap = Utils.generateIcon(context, group, true).getLetterTile();
+
+        IconCompat icon = IconCompat.createWithAdaptiveBitmap(iconBitmap);
+
+        return new ShortcutInfoCompat.Builder(context, group._id)
```

Comment:
> I just added the following: an error message now appears if the group name has changed or has been deleted. That's the best thing I could do for this case :)

You should implement an [updateShortcuts](https://developer.android.com/reference/androidx/core/content/pm/ShortcutManagerCompat#updateShortcuts(android.content.Context,%20java.util.List%3Candroidx.core.content.pm.ShortcutInfoCompat%3E)) function for groups. I believe it uses the group name here (`group._id`). But it would be nicer if you would prefix the shortcut names to make them clearly different from loyalty cards (like `GROUP_groupname`). You will have to call that function whenever a group is edited or deleted.

Reply:
That's not what I meant. I meant it like this:

If group._id = Fashion, don't use "Fashion" as shortcut ID, but use "GROUP_Fashion". That way you can tell which shortcuts are group-related in the code easily because they all start with "GROUP_". That makes searching for the right shortcut in the updateShortcuts function way easier (no possibility to get confused with cards).

Reply:
I see what you mean. Indeed I already tried to implement a updateShortcuts and removeShortchuts functions for group shortcuts, but nothing was happening with my change: the name of the shortcut doesn't change and the shortcut doesn't get deleted. This is also the same for card shortcuts on my android 5.0.2 device, could this issue be related to a too low api level?

Reply:
I do get comment notifications, sorry, been a bit busy.

>Does the updateShortcut and removeShortcut work for you with cards, e.g. is the shortcut title and icon updated/removed if the card is updated/removed?

updating works, but removing doesn't. I've done this:
1. Create a shortcut for a card named "Clothing store"
2. Rename the card to "test" -> shortcut changed
3. Changed the card image -> shortcut changed
4. Removed the shortcut -> no change

I'll make a new GitHub issue for that, the deletion should do something. Created #686 for that.

Reply:
> I do get comment notifications, sorry, been a bit busy.

That's normal, no problem :)

> * Create a shortcut for a card named "Clothing store"
> 
> * Rename the card to "test" -> shortcut changed
> 
> * Changed the card image -> shortcut changed

This doesn't work on Android 5.0.2 device, so I will open an issue for that (probably related to the old api level). I can try to implement updateShortcut for group shortcuts, but I will not be able to test it on my device :/ So should I implement this in this PR?

Reply:
I think I first want to see if we can fix this not working for cards on Android 5, if that can be fixed it could be implemented here too. Otherwise I will review it without that feature and add it myself for newer Android versions.